src/devices/ovrheadset/OVRHeadset.cpp

```diff
@@ -151,19 +155,167 @@ yarp::dev::OVRHeadset::OVRHeadset() :
+    buttonIdToOvrButton.push_back(ovrButton_B);
+    buttonIdToOvrButton.push_back(ovrButton_RThumb);
+    buttonIdToOvrButton.push_back(ovrButton_RShoulder);
+    buttonIdToOvrButton.push_back(ovrButton_X);
+    buttonIdToOvrButton.push_back(ovrButton_Y);
+    buttonIdToOvrButton.push_back(ovrButton_LThumb);
+    buttonIdToOvrButton.push_back(ovrButton_LShoulder);
+    buttonIdToOvrButton.push_back(ovrButton_Enter);
+    buttonIdToOvrButton.push_back(ovrButton_Back);
+    buttonIdToOvrButton.push_back(ovrButton_VolUp);
+    buttonIdToOvrButton.push_back(ovrButton_VolDown);
+    buttonIdToOvrButton.push_back(ovrButton_Home);
+}
+
+void yarp::dev::OVRHeadset::fillHatStorage()
+{
+    DButtonToHat[0]               = YRPJOY_HAT_CENTERED;
+    DButtonToHat[ovrButton_Up]    = YRPJOY_HAT_UP;
+    DButtonToHat[ovrButton_Right] = YRPJOY_HAT_RIGHT;
+    DButtonToHat[ovrButton_Down]  = YRPJOY_HAT_DOWN;
+    DButtonToHat[ovrButton_Left]  = YRPJOY_HAT_LEFT;
+}
+
 bool yarp::dev::OVRHeadset::open(yarp::os::Searchable& cfg)
 {
     yTrace();
+    //PORT OPENING CODE REFACTORING PROPOSAL
+    //std::vector<std::pair<yarp::os::BufferedPort<yarp::os::Bottle>*, std::string> > ports;
+    //ports.push_back(std::make_pair(orientationPort,                  "orientation"));
+    //ports.push_back(std::make_pair(positionPort,                     "position"));
+    //ports.push_back(std::make_pair(angularVelocityPort,              "angularVelocity"));
+    //ports.push_back(std::make_pair(linearVelocityPort,               "linearVelocity"));
+    //ports.push_back(std::make_pair(angularAccelerationPort,          "angularAcceleration"));
+    //ports.push_back(std::make_pair(linearAccelerationPort,           "linearAcceleration"));
+    //ports.push_back(std::make_pair(predictedOrientationPort,         "predictedOrientation"));
+    //ports.push_back(std::make_pair(predictedPositionPort,            "predictedPosition"));
+    //ports.push_back(std::make_pair(predictedAngularVelocityPort,     "predictedAngularVelocity"));
+    //ports.push_back(std::make_pair(predictedLinearVelocityPort,      "predictedLinearVelocity"));
+    //ports.push_back(std::make_pair(predictedAngularAccelerationPort, "predictedAngularAcceleration"));
+    //ports.push_back(std::make_pair(predictedLinearAccelerationPort,  "predictedLinearAcceleration"));
+    //
+    //for (auto port : ports)
+    //{
+    //    port.first = new yarp::os::BufferedPort<yarp::os::Bottle>;
+    //    std::string name, prefix;
+    //    bool        predicted;
+    //
+    //    predicted = port.second.find("predicted") != std::string::npos;
+    //    prefix = predicted ? "/oculus/predicted" : "/oculus";
+    //    name = prefix + "/headpose/" + port.second + ":o";
+    //    if (!port.first->open(name)) {
+    //        yError() << "Cannot open" << port.second << "port";
+    //        this->close();
+    //        return false;
+    //    }
+    //    port.first->setWriteOnly();
+    //}
```

Comment:
Please, leave this comment out of the PR, since it is not related...
I'm partially ok with this refactoring (I'd use `std::map` instead of `std::vector<std::pair>` and `emplace` instead of `push_back`), but we can discuss this on a separate PR